src/js/control/autocomplete.js

```diff
@@ -192,22 +211,78 @@ export default class controlAutocomplete extends control {
    * @param {Object} selectedOption - option - 'li' element - to be selected in autocomplete list
    */
   selectOption(list, selectedOption) {
-    const options = list.querySelectorAll('li');
-    options.forEach((option)=>{
-      option.classList.remove('active-option');
-    });
+    const options = list.querySelectorAll('li'); 
+    // --Fix for IE11
+    for (let i = 0; i < options.length; i++) {
+      options[i].classList.remove('active-option');
+    }
     if (selectedOption) {
       selectedOption.classList.add('active-option');
     }
   }
+   
 
   /**
-   * When the element is rendered into the DOM, execute the following code to initialise it
-   * @param {Object} evt - event
+   * Is the value in the autocomplete field in the pre-defined Options list?
+   * @param {Object} list - list of autocomplete options
+   * @param {Object} value -value trying to be set
+   * @return {Object} - is the option in the pre defined list
    */
+  isOptionValid(list,value){
+    const options = list.querySelectorAll('li'); 
+    let validValue = false;
+    for (let i = 0; i < options.length; i++) {
+      if(options[i].innerHTML === value){
+        validValue = true;
+        break;
+      }
+    }
+    return validValue;  
+  }
+  
+  /**
+   * onRender callback
+   * @param {Object} evt
+   */  
   onRender(evt) {
+    // Set userData if available
+    if(this.config.userData){  
+      let id = $('#'+this.config.name).attr('id');
+      let preSelectedOption = this.config.userData[0];
+ 
+      const list = document.getElementById(id).nextSibling;
+      let selectedOption= null;
+
+      $('#' +id + '-list' + ' li').each(function(){   
+      // eslint-disable-next-line no-invalid-this             
+        if($(this).attr('value') === preSelectedOption){
+       // eslint-disable-next-line no-invalid-this          
+          selectedOption = $(this).get(0);                             
+        }
+      }); 
+     // If the option was not set, and configuration says it doesn't have to be pre-defined, set the value
+     if(selectedOption === null){
+       if(this.config.requireValidOption){
+         // Don't allow
+         return;
+       }else{
+         // Set it to whatever the value is
+        $('#' + id).prev().val(this.config.userData[0]);
+        return;         
+       }
+     }     
+
+      $('#' + id).prev().val(selectedOption.innerHTML);
+      $('#' + id).val(selectedOption.getAttribute('value'));
+      
+      if (list.style.display === 'none') {
+        this.showList(list, selectedOption);
```

Comment:
where do `showList` and `hideList` come from?

Reply:
These already existed further up in the file. ~ lines 150-170

Reply:
ah ok. i didnt see in PR diff and its been a while since i wrote that :smile:

Reply:
Ok I updated this. Seems like it is more performant